domains/pallets/messenger/src/lib.rs

```diff
@@ -1399,6 +1389,19 @@ mod pallet {
             UpdatedChannels::<T>::get()
         }
 
+        pub fn open_channels() -> BTreeSet<(ChainId, ChannelId)> {
+            Channels::<T>::iter().fold(
+                BTreeSet::new(),
+                |mut acc, (dst_chain_id, channel_id, channel)| {
+                    if channel.state != ChannelState::Closed {
+                        acc.insert((dst_chain_id, channel_id));
+                    }
+
+                    acc
+                },
+            )
+        }
```

Comment:
Optional: if we're concerned about performance with large numbers of open channels, this might make more efficient use of memory and CPU (or it might not, because it performs an extra `collect::<Vec>` and `sort`).

```rust
Channels::<T>::iter()
    .filter_map(|(dst_chain_id, channel_id, channel)| {
        if channel.state != ChannelState::Closed {
            Some((dst_chain_id, channel_id))
        } else {
            None
        }
    })
    .collect::<BTreeSet<_>>()
```

Reply:
I would leave it as is